iothub/service/src/Messaging/Models/OutgoingMessage.cs

```diff
@@ -221,12 +221,17 @@ public DeliveryAcknowledgement Ack
         internal ArraySegment<byte> DeliveryTag { get; set; }
 
         /// <summary>
-        /// Gets the payload as a byte array.
+        /// Gets the payload as a byte array, serialized and encoded if necessary.
         /// </summary>
-        /// <returns>A fully encoded serialized string as bytes.</returns>
+        /// <remarks>
+        /// If needed, serialization uses Newtonsoft.Json and encoding is UTF8.
+        /// </remarks>
+        /// <returns>A payload as a byte array.</returns>
         internal byte[] GetPayloadObjectBytes()
         {
-            return Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(Payload));
+            return Payload is byte[] payloadAsByteArray
+                ? payloadAsByteArray
+                : Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(Payload));
```

Comment:
Not for this change specifically, but do you remember what our test coverage looks like for different payload types for each of the operations (send telemetry, receive incoming message, methods, twins etc.)?

Reply:
For C2D, it looks like we pretty much just send a string or an object for all the tests.

Reply:
I added a test for this specific case.